modules/HumanStateVisualizer/src/main.cpp

```diff
@@ -0,0 +1,259 @@
+        return EXIT_FAILURE;
+    }
+
+    // wait for the iHumanState to be initialized
+    while (iHumanState->getBaseName().empty())
+    {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        yInfo() << LogPrefix << "Waiting for data from HumanStateRemapper";
+    }
+     
+    // compare the iHumanState base and joint names with the visualization model
+    yInfo() << LogPrefix << "Human State Interface providing data for the base link [ " << iHumanState->getBaseName() << " ]";
+    if ( iHumanState->getBaseName() != viz.modelViz("human").model().getLinkName(viz.modelViz("human").model().getDefaultBaseLink()))
+    {
+        viz.modelViz("human").model().setDefaultBaseLink(viz.modelViz("human").model().getLinkIndex(iHumanState->getBaseName()));
+        yInfo() << LogPrefix << "Defaul base link of the visualized model is changed to " << iHumanState->getBaseName();
+    }
+    yInfo() << LogPrefix << "Human State Interface providing data from [ " << iHumanState->getJointNames().size() << " ] joints";
+    
+    for (auto jointName : iHumanState->getJointNames())
+    {
+        if (viz.modelViz("human").model().getJointIndex(jointName) == iDynTree::JOINT_INVALID_INDEX)
+        {
+            if (!ignoreMissingLinks)
+            {
+                yError() << LogPrefix << "joint [ " << jointName << " ] not found in the visualized model." 
+                         << " Set ignoreMissingLinks true if you want to ignore the missing joints.";
+                return EXIT_FAILURE;
+            }
+            yWarning() << LogPrefix << "joint [ " << jointName << " ] not found in the visualized model, "
+                       << "the joint will be ignored.";
+        }
+    }
+
+    // initialize state variables for visualization
+    iDynTree::Transform wHb = iDynTree::Transform::Identity();
+    iDynTree::VectorDynSize joints(viz.modelViz("human").model().getNrOfDOFs());
+    joints.zero();
+
+    // start Visualization
+    std::chrono::steady_clock::time_point now = std::chrono::steady_clock::now();
+    std::chrono::steady_clock::time_point lastViz = std::chrono::steady_clock::now();
+
+    long minimumMicroSecViz = std::round(1e6 / (double) maxVisualizationFPS);
+
+    while(viz.run())
+    {
+        now = std::chrono::steady_clock::now();
+        if (std::chrono::duration_cast<std::chrono::microseconds>(now - lastViz).count() < minimumMicroSecViz)
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
+            continue;
+        }
+
+        // read values from iHumanState
+        std::array<double, 3> basePositionInterface = iHumanState->getBasePosition();
+        std::array<double, 4> baseOrientationInterface = iHumanState->getBaseOrientation();
+        std::vector<double> jointPositionsInterface = iHumanState->getJointPositions();
+        std::vector<std::string> jointNames = iHumanState->getJointNames();

```

Comment:
If you declare these variables before, you may avoid allocating memory at every loop.

Reply:
Done in https://github.com/robotology/human-dynamics-estimation/pull/230/commits/a1c4a63c8b786ec8ff2d0c29d745f1f4169ec63d (I took the chance to improve the same lines in `HumanStateWrapper` device)